hibernate-core/src/main/java/org/hibernate/type/CollectionType.java

```diff
@@ -654,7 +654,12 @@ public Object replace(
 			final Object owner,
 			final Map<Object, Object> copyCache) throws HibernateException {
 		if ( original == null ) {
-			return null;
+			if ( target == null  ) {
+				return target;
+			}
+			final Collection collection = (Collection) target;
+			collection.clear();
```

Comment:
Not sure if clearing the original collection is always possible. Maybe it's better to return a fresh `PersistentCollection`?

Reply:
I tied but returning a fresh `PersistentCollection` causes few exceptions ` A collection with orphan deletion was no longer referenced by the owning entity instance`
I think `target` is always a `PersistentCollection` so it should be safe to call `clear()`

Reply:
If that is the case, we should IMO add an assert statement here to make sure.